src/wp-admin/includes/update-core.php

```diff
@@ -1574,6 +1577,52 @@ function _preload_old_requests_classes_and_interfaces( $to ) {
 	}
 }
 
+/**
+ * Preloads old classes and interfaces required for updating Core.
+ *
+ * This function preloads the old code into memory before the
+ * upgrade process deletes the files. These files are loaded into
+ * memory via an autoloader, meaning when a class or interface is needed
+ * WordPress Core could attempt to access code. If the file is not there,
+ * a fatal error could occur. If the file was replaced, the new code is not
+ * compatible with the old, resulting in a fatal error.
+ * Preloading ensures the code is in memory before the code is updated.
+ *
+ * @since 6.6.0
+ */
+function _preload_core_classes_and_interfaces() {
```

Comment:
For this it would be good to have some tests to make sure it's kept up to date (I am unsure what these look like, sorry).

For the upgrade routine, it would be good to test this PR for:
* successful upgrade
* failed upgrade/rollback 

cc @afragen @costdev